### PR TITLE
Remove V2 API usage from Integration tests

### PR DESCRIPTION
**Pull request by serathius**
Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

cc @ptabor 
Required for https://github.com/etcd-io/etcd/pull/13612 and https://github.com/etcd-io/etcd/issues/12913

### CODE REVIEW

File: tests/integration/cluster_test.go [hunk 1]
```diff
@@ -74,8 +72,7 @@ func TestClusterOf3UsingDiscovery(t *testing.T) { testClusterUsingDiscovery(t, 3
 
 func testClusterUsingDiscovery(t *testing.T, size int) {
 	integration.BeforeTest(t)
-	dc := integration.NewCluster(t, 1)
-	dc.Launch(t)
+	dc := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 1, UseIP: true})
 	defer dc.Terminate(t)
 	// init discovery token space
 	dcc := integration.MustNewHTTPClient(t, dc.URLs(), nil)
```
Comment (ptabor) [inline tests/integration/cluster_test.go:78]: What's our strategy with discovery (testing). 

Migrate to e2e test with loading 3.5 binary as a discovery server ?

File: tests/integration/client/client_test.go [hunk 2]
```diff
@@ -1,130 +0,0 @@
-// Copyright 2016 The etcd Authors
```
Comment (ptabor) [inline tests/integration/client/client_test.go:1]: Can we delete it without explicit decision about removing the discovery protocol, 
as the client probably is being used to register during the discovery process. 

Depending on v3.5 client module from v3.6 server code might create conflicts on overlapping parts of code
(but is also path to explore).

File: tests/framework/integration/cluster.go [hunk 3]
```diff
@@ -598,6 +601,7 @@ type Member struct {
 
 	// ServerClient is a clientv3 that directly calls the etcdserver.
 	ServerClient *clientv3.Client
+	Client       *clientv3.Client
```
[the remark this answers is not in the document]
Comment (serathius) [reply tests/framework/integration/cluster.go:554]: Done

File: tests/integration/cluster_test.go (hunk 1 above)
Comment (serathius) [reply tests/integration/cluster_test.go:78]: SG

File: tests/integration/client/client_test.go (hunk 2 above)
Comment (serathius) [reply tests/integration/client/client_test.go:1]: Not sure what you mean, should client integration tests remain or not? For discovery protocol I think that we will have e2e tests.

**Comment by serathius** [comment]
Based on this PR we should add `Reimplement v2 discovery protocol on v3 api` in https://github.com/etcd-io/etcd/issues/12913

Consequences:
* v3.5 cluster will not be able to boostrap on v3.6 cluster as it will not support V2 API.
* v3.5 cluster can boostrap on v3.5 with `--enable-v2` flag or v3.4 cluster using V2 API.
* v3.6 cluster will still be able to bootstrap on v3.4, v3.5, v3.6 clusters using V3 API.

**Comment by ptabor** [comment]
> Based on this PR we should add `Reimplement v2 discovery protocol on v3 api` in #12913
> 
> Consequences:
> 
> * v3.5 cluster will not be able to boostrap on v3.6 cluster as it will not support V2 API.
> * v3.5 cluster can boostrap on v3.5 with `--enable-v2` flag or v3.4 cluster using V2 API.
> * v3.6 cluster will still be able to bootstrap on v3.4, v3.5, v3.6 clusters using V3 API.

I'm not sure if this is a problem, but want to call it explicitly: 
 * discovery service will not be usable for upgrading v3.5 -> v3.6 as V2 api and V3 api use different storage within etcd 
   so the protocol change means also 'independent' state unless discovery service is using `--enable-v2v3'.

**Review by ptabor — APPROVED** [review]

**Review by spzala — APPROVED** [review]
LGTM Thanks @serathius

**Pull request merged**